### PR TITLE
Major re-write

### PR DESCRIPTION
**Pull request by martin9700**
Hi Joe, really liked the module but wanted to make just a small change or two (joke).

Changed branding to BitbucketPS everywhere I found it.  Add BB to the cmdlet's to try and avoid collisions with other PS modules.  Added a BBSession global variable to store credentials and URL's to make using the module a lot easier (pretty standard practice).  Then a ton of functional changes to make everything work and look good.  

All original functions were either modified or removed (many were combined).  

Hope a big re-write like this is OK!

### CODE REVIEW

**Comment by lipkau** [comment]
Great stuff! Thanks for the contribution.

I will review it this afternoon

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Ignore config file
 config.xml
+BBconfig.xml
```
Comment (lipkau) [inline .gitignore]: why would this be needed?

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,7 @@
...
+*exclude*
```
Comment (lipkau) [inline .gitignore]: This kind of ignores should be handles locally

File: BitbucketPS.psd1 [hunk 3]
```diff
@@ -0,0 +1,131 @@
+#
```
Comment (lipkau) [inline BitbucketPS.psd1]: Don't create the file again.
Just save it as `UTF-8` instead of `UTF-16 BE`

File: BitbucketPS.psm1 [hunk 4]
```diff
@@ -0,0 +1,489 @@
+function ConvertFrom-Json2
```
Comment (lipkau) [inline BitbucketPS.psm1]: What is going on here?

Why are the functions being declared in the psm1 file?
psm1 file was generic already (but should be updated to this: https://github.com/lipkau/JiraPS/blob/enhance/harmonizeStyling/JiraPS/JiraPS.psm1)

File: Build_exclude.ps1 [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿$ModuleInformation = @{
```
Comment (lipkau) [inline Build_exclude.ps1]: This should definitely not be in here.
This is specific to your dev environment.

**Also, AtlassianPS uses `InvokeBuild`**

File: Import-BitBucket_exclude.ps1 [hunk 6]
```diff
@@ -0,0 +1,17 @@
+﻿$Path = "C:\Dropbox\athenaGIT\BitbucketPS"
```
Comment (lipkau) [inline Import-BitBucket_exclude.ps1]: $Path should be generic and the rest should be in the loading of the module (`.psm1`)

**Review by lipkau — CHANGES_REQUESTED** [review]
Bellow are a lot of changes required.

Also, `Old_exclude` should not be in here

**Let me know when you are done, or if you will take a while for this. I want to make some changes to the repo that will case a lot of merge conflicts**

File: BitbucketPS.psm1 (hunk 4 above)
Comment (martin9700) [reply BitbucketPS.psm1]: I dislike that approach, it causes slower module loads especially if someone places the module on a network share.  By putting the functions in the PSM file they load almost instantly.

File: Build_exclude.ps1 (hunk 5 above)
Comment (martin9700) [reply Build_exclude.ps1]: That's fine, the gitignore should have stopped these files from going up there.

File: Import-BitBucket_exclude.ps1 (hunk 6 above)
Comment (martin9700) [reply Import-BitBucket_exclude.ps1]: Again, shouldn't have made it into the repo, not sure why gitignore ignored it (see what I did there)

File: Build_exclude.ps1 (hunk 5 above)
Comment (lipkau) [reply Build_exclude.ps1]: you need to commit .gitignore with the changes before commit the files it is supposed to ignore.
you need to revert gitignore and commit the deletion of those files... then you can add it to gitignore again...

**however, please use this for stuff that is not part of the repo, but custom to your setup:**
https://www.peterbe.com/plog/gitinfoexclude

File: BitbucketPS.psm1 (hunk 4 above)
Comment (lipkau) [reply BitbucketPS.psm1]: I understand your preference.
However, this is a community managed repo and we have established the loading of dedicated files as the standard in order to make the maintenance and readability easier.

File: Build_exclude.ps1 (hunk 5 above)
Comment (martin9700) [reply Build_exclude.ps1]: Perfect, thank you for the link.

File: BitbucketPS.psd1 [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
 #
-# Generated on: 1/13/2018
+# Generated on: 12/6/2016
 #
 
 @{
 
 # Script module or binary module file associated with this manifest.
-RootModule = 'BitbucketPS'
+RootModule = '.\PSBitBucket.psm1'
```
Comment (lipkau) [inline BitbucketPS.psd1]: My bad.
This change was valid. Please restore

File: BitbucketPS.psd1 [hunk 8]
```diff
@@ -18,22 +18,22 @@ ModuleVersion = '1.0'
...
 # ID used to uniquely identify this module
-GUID = 'fdc27852-7527-4771-a72c-680746446df8'
+GUID = 'd859fc87-50ef-406a-98a9-0355d55d1d4d'
 
 # Author of this module
 Author = 'Joe Beaudry'
 
 # Company or vendor of this module
-CompanyName = 'Atlassian'
+CompanyName = 'Unknown'
```
Comment (lipkau) [inline BitbucketPS.psd1]: My bad.
This change was valid. Please restore

File: BitbucketPS.psd1 [hunk 9]
```diff
@@ -18,22 +18,22 @@ ModuleVersion = '1.0'
...
 
 # Copyright statement for this module
-Copyright = '(c) 2018 Joe Beaudry. All rights reserved.'
+Copyright = '(c) 2016 Joe Beaudry. All rights reserved.'
```
Comment (lipkau) [inline BitbucketPS.psd1]: My bad.
This change was valid. Please restore

File: BitbucketPS.psd1 [hunk 10]
```diff
@@ -18,22 +18,22 @@ ModuleVersion = '1.0'
...
 
 # Description of the functionality provided by this module
-Description = 'PowerShell module to interact with Atlassian Bitbucket '
+# Description = ''
```
Comment (lipkau) [inline BitbucketPS.psd1]: My bad.
This change was valid. Please restore

File: BitbucketPS.psd1 [hunk 11]
```diff
@@ -18,22 +18,22 @@ ModuleVersion = '1.0'
...
 
 # Minimum version of the Windows PowerShell engine required by this module
-PowerShellVersion = '3.0'
+# PowerShellVersion = ''
```
Comment (lipkau) [inline BitbucketPS.psd1]: My bad.
This change was valid. Please restore

File: BitbucketPS.psd1 [hunk 12]
```diff
@@ -99,27 +98,20 @@ PrivateData = @{
         # Tags = @()
 
         # A URL to the license for this module.
-        LicenseUri = 'https://github.com/AtlassianPS/BitbucketPS/blob/master/LICENSE'
+        # LicenseUri = ''
```
Comment (lipkau) [inline BitbucketPS.psd1]: My bad.
This change was valid. Please restore

File: BitbucketPS.psd1 [hunk 13]
```diff
@@ -99,27 +98,20 @@ PrivateData = @{
         # Tags = @()
 
         # A URL to the license for this module.
-        LicenseUri = 'https://github.com/AtlassianPS/BitbucketPS/blob/master/LICENSE'
+        # LicenseUri = ''
 
         # A URL to the main website for this project.
-        ProjectUri = 'https://github.com/AtlassianPS/BitbucketPS'
+        # ProjectUri = ''
```
Comment (lipkau) [inline BitbucketPS.psd1]: My bad.
This change was valid. Please restore

File: BitbucketPS.psd1 [hunk 14]
```diff
@@ -99,27 +98,20 @@ PrivateData = @{
...
         # ReleaseNotes of this module
-        ReleaseNotes = 'Updating gitignore
-
-Incorporated the rest URI into the BBSession varialbe. Changed that URI to use /latest/ instead of /1.0/ (unless Atlassian tells me that''s a bad idea).  Updated output on BBCommits, BBRepositories to look a little prettier
-
-'
-
-        # External dependent modules of this module
-        # ExternalModuleDependencies = ''
+        # ReleaseNotes = ''
```
Comment (lipkau) [inline BitbucketPS.psd1]: This shall contain the link to the `CHANGELOG.md` in the root of the repo.
Feel free to create an empty (or not empty) file

**Comment by lipkau** [comment]
@martin9700, are you ok with me pushing some commits to your PR?

**Review by lipkau — CHANGES_REQUESTED** [review]

File: BitbucketPS.psd1 (hunk 3 above)
Comment (martin9700) [reply BitbucketPS.psd1]: Back to original

**Comment by martin9700** [comment]
Absolutely

File: BitbucketPS.psd1 (hunk 14 above)
Comment (martin9700) [reply BitbucketPS.psd1]: That works, I reverted so it should be back to blank.  Can be updated once this mess of a pull request is done!

**Comment by lipkau** [comment]
Don't forget to pull before continuing to code

**Comment by lipkau** [comment]
I would like to close this PR.
I am pushing changes that are aligned with the other modules.
Please update your fork and use the develop branch.

This is what I am working on
https://github.com/lipkau/BitbucketPS/tree/feature/%2320-SessionHandling

**Pull request closed without merging**